From f2c6ccb102db1fa1b832b0df16da4dc998ce14b7 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Thu, 25 Jan 2018 16:23:25 +0800
Subject: [PATCH 3259/5242] MLK-17461-4: dts: reparent dpu/hdmi pixel clock
 from av_pll_bypass

commit  a621ca13b40ed595a70e9d768cb3aef8f5a0d825 from
https://source.codeaurora.org/external/imx/linux-imx.git

Reparent dpu and hdmi pixel clock from av_pll_bypass.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts     |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
index be4e300..ad4df52 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
@@ -31,6 +31,13 @@
 	};
 };
 
+&dpu1 {
+		assigned-clocks = <&clk IMX8QM_DC0_DISP0_SEL>,
+						<&clk IMX8QM_DC0_DISP1_SEL>;
+		assigned-clock-parents = <&clk IMX8QM_DC0_BYPASS_0_DIV>,
+						<&clk IMX8QM_DC0_PLL1_CLK>;
+};
+
 &ldb1_phy {
 	status = "disabled";
 };
@@ -45,6 +52,12 @@
 
 &hdmi {
 	compatible = "fsl,imx8qm-hdmi";
+	assigned-clocks = <&clk IMX8QM_HDMI_PXL_SEL>,
+					<&clk IMX8QM_HDMI_PXL_LINK_SEL>,
+					<&clk IMX8QM_HDMI_PXL_MUX_SEL>;
+	assigned-clock-parents = <&clk IMX8QM_HDMI_AV_PLL_BYPASS_CLK>,
+							<&clk IMX8QM_HDMI_AV_PLL_BYPASS_CLK>,
+							<&clk IMX8QM_HDMI_AV_PLL_BYPASS_CLK>;
 	status = "okay";
 };
 
-- 
1.7.9.5

